draft-ietf-ccamp-dwdm-if-param-yang.xml

```diff
@@ -526,191 +612,343 @@ module ietf-wdm-interface {
+    base wdm-if-tca-type;
+    description
+      "Counter of Post-FEC uncorrected words TCA on the receiver
+      interface";
+  }
+
+  identity rx-q-factor-tca{
+    base wdm-if-tca-type;
+    description "Q-Factor TCA on the receiver interface";
   }
 
 
   grouping wdm-if-tca-thresholds {
     description "Thresholds for TCA's";
     leaf tca-type {
-      type wdm-if-tca-types;
+      type identityref {
+        base wdm-if-tca-type;
+      }
+      mandatory true;
       description
         "type of the TCA that identify the
-        performance measurement, eg TX Power";
+        performance measurement, eg tx-power-tca";
     }
     leaf tca-name {
       type string;
+      mandatory true;
       description
-        "A name of the TCA that explain the TCA scope
-        eg 'Low TX Power'";
+        "A textual name of the TCA that explain its scope
+        eg 'TX Power Degrade'";
     }
     leaf raise-threshold {
-      type int32;
+      type l0-types:decimal-5;
+      mandatory true;
       description
         "A TCA is raised if the variable cross this threshold:
         - if raise-threshold is greater than clear-threshold
         (or clear threshold is not defined), the TCA
         is raised when the value exceed the threshold
         - if raise-threshold is smaller that clear-threshold, the
-        TCA is raised when the value fall below the threshold";
+        TCA is raised when the value fall below the threshold
+        - if clear-threshold is not defined the TCA raise and clear
+        on the same raise-threshold";
     }
     leaf clear-threshold {
-      type int32;
+      type l0-types:decimal-5;
       description
         "A TCA cleared when the variable cross the threshold in the
-        direction defined for raise-threshold";
+        direction defined for raise-threshold.
+        If clear-threshold is not defined the TCA raise and cleare
+        on the same raise-threshold";
     }
-  }
+  }  // end grouping wdm-if-tca-threshold
 
   grouping wdm-if-tca-list {
     description "List of TCA's";
     leaf number-of-tcas-supported {
       type uint32;
+      default 0;
       config false;
-      description "Number of TCAs supported by this interface";
+      description
+        "Number of TCAs supported by this interface,
+        if the implementation do not suport TCAs it SHALL leave this
+        value to defualt = 0";
     }
     list tca-list {
-      key "tca-type";
+      key "tca-id";
       description "List of the TCAs";
+      leaf tca-id {
+        type string;
+        mandatory true;
+        description
+          "Unique Identifier of the TCA defined for the interface";
+      }
       uses wdm-if-tca-thresholds;
     }
-  }
+  }  // end grouping wdm-if-tca-list
 
-  grouping wdm-if-mode-params {
-    description "OCh mode parameters";
+  grouping wdm-if-config {
+    description "DWDM Interface Configuration parameters";
 
-    leaf min-osnr-margin {
```

Comment:
This leaf was supposed to be the min-OSNR for the operation of the mode, i.e. the FEC threshold. It is used by GNPy and other simulations to calculate the Quality of Transmission. It should therefore be kept, probably under the name "min-OSNR" to avoid misunderstandings with the if-statistics where "min-OSNR-margin" reports only the margin to this reference.